plugins/transformations/sequence2array/srcCxx/Sequence2ArrayTransformation.cxx

```diff
@@ -212,11 +190,11 @@ bool Seq2ArrayTransformation::are_type_compatible(
             case TypeKind::SEQUENCE_TYPE:
             case TypeKind::ARRAY_TYPE:
             case TypeKind::UNION_TYPE:
-                is_compatible = is_compatible && are_type_compatible(
+                is_compatible = is_compatible && are_types_compatible(
```

Comment:
I just realized there is a problem with the implementation of `are_types_compatible()`.

The function is inconsistent in that it returns a boolean flag to inform the user whether two types are compatible, but at the same time it throws exceptions as soon as it detects an incompatible type, instead of returning `false`.

This makes every instance where `are_types_compatible()` is called and the returned value is then checked completely redundant, since the function can effectively only return `true` or throw an exception.

You should pick either one of the approaches and update the code to accordingly:

- If you go with returning the boolean flag, I would convert any instance of an exception with a log message and an immediate `return false`.
- If you want to maintain the exception-based control flow, then I suggest refactoring `are_types_compatible()` to something like `void assert_types_compatible()` which throws an exception in case of incompatibility. If you go with this strategy, then I would recommend to define your own exception type instead of using `std::runtime_error` so that you may intercept these exceptions and handle them while they are "bubbling up" to routing service, without risking hiding some runtime_error you didn't generate.